core/src/main/scala/com/evolutiongaming/kafka/flow/registry/EntityRegistry.scala

```diff
@@ -0,0 +1,87 @@
+package com.evolutiongaming.kafka.flow.registry
+
+import cats.effect.Resource
+import cats.effect.kernel.Ref
+import cats.syntax.all._
+import cats.{Applicative, Monad}
+
+/** Observability API allowing to inspect current state of in-memory entities externally (from HTTP handler, for example).
+  * When passed to the library, it's used to register an association between entity key and the function
+  * to obtain its current state when the state is initialized for the first time.
+  * When the entity is evicted from memory, this association is also removed.
+  *
+  * There are three pre-defined implementations currently provided:
+  *   - `EntityRegistry.empty` for no-op implementation
+  *   - `EntityRegistry.const` for immutable constant data
+  *   - `EntityRegistry.memory`, fully functional in-memory registry
+  * @tparam K entity key
+  * @tparam S entity state
+  */
+trait EntityRegistry[F[_], K, S] {
+
+  /** Registers an association between an entity's key and the function to obtain current state of it.
+    * This is used internally by the library and shouldn't be called by the library users.
+    * @param key entity key
+    * @param state computation (effectively function) to obtain the current value of the entity
+    * @return
+    */
+  def register(key: K, state: F[Option[S]]): Resource[F, Unit]
```

Comment:
why is this a resource? you add to scaladoc that freeing this resource removes the entity from registry

Reply:
It says __When the entity is evicted from memory, this association is also removed.__ Did you mean something else?

Reply:
>this means that the association is removed when the entity is garbage collected

This is not what's meant actually. It means that it's going to be removed when we remove the entity from the internal cache. I'll rephrase this.

Reply:
Done, please see if it's phrased clearer now